<a href="https://colab.research.google.com/github/manusaleal/EDA_Projeto_Gas/blob/desenvolvimento-cicero/EDA_Projeto_Gas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Projeto

In [333]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [334]:
import pandas as pd
import numpy as np

In [335]:
#Abrindo a base de dados .csv
df_base = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/TPI (PY) - Projeto Final/2004-2021.csv')

#Realizando uma rápida observação dos dados
df_base.head(10)

DATA INICIAL  DATA FINAL        REGIÃO              ESTADO  \
0   2004-05-09  2004-05-15  CENTRO OESTE    DISTRITO FEDERAL   
1   2004-05-09  2004-05-15  CENTRO OESTE               GOIAS   
2   2004-05-09  2004-05-15  CENTRO OESTE         MATO GROSSO   
3   2004-05-09  2004-05-15  CENTRO OESTE  MATO GROSSO DO SUL   
4   2004-05-09  2004-05-15      NORDESTE             ALAGOAS   
5   2004-05-09  2004-05-15      NORDESTE               BAHIA   
6   2004-05-09  2004-05-15      NORDESTE               CEARA   
7   2004-05-09  2004-05-15      NORDESTE            MARANHAO   
8   2004-05-09  2004-05-15      NORDESTE             PARAIBA   
9   2004-05-09  2004-05-15      NORDESTE          PERNAMBUCO   

            PRODUTO  NÚMERO DE POSTOS PESQUISADOS UNIDADE DE MEDIDA  \
0  ETANOL HIDRATADO                           127              R$/l   
1  ETANOL HIDRATADO                           387              R$/l   
2  ETANOL HIDRATADO                           192              R$/l   
3  ETANOL HIDRATADO                           162              R$/l   
4  ETANOL HIDRATADO                           103              R$/l   
5  ETANOL HIDRATADO                           408              R$/l   
6  ETANOL HIDRATADO                           278              R$/l   
7  ETANOL HIDRATADO                           105              R$/l   
8  ETANOL HIDRATADO                           125              R$/l   
9  ETANOL HIDRATADO                           423              R$/l   

   PREÇO MÉDIO REVENDA  DESVIO PADRÃO REVENDA  PREÇO MÍNIMO REVENDA  \
0                1.288                  0.016                 1.190   
1                1.162                  0.114                 0.890   
2                1.389                  0.097                 1.180   
3                1.262                  0.070                 1.090   
4                1.181                  0.078                 1.050   
5                1.383                  0.132                 0.999   
6                1.453                  0.218                 1.030   
7                1.631                  0.158                 1.350   
8                1.284                  0.130                 1.100   
9                1.224                  0.141                 0.989   

   PREÇO MÁXIMO REVENDA MARGEM MÉDIA REVENDA  COEF DE VARIAÇÃO REVENDA  \
0                 1.350                0.463                     0.012   
1                 1.449                0.399                     0.098   
2                 1.760                0.419                     0.070   
3                 1.509                0.432                     0.055   
4                 1.400                 0.24                     0.066   
5                 2.050                0.426                     0.095   
6                 1.950                0.353                     0.150   
7                 2.000                0.515                     0.097   
8                 1.699                0.353                     0.101   
9                 1.700                0.277                     0.115   

  PREÇO MÉDIO DISTRIBUIÇÃO DESVIO PADRÃO DISTRIBUIÇÃO  \
0                    0.825                       0.11   
1                    0.763                      0.088   
2                     0.97                      0.095   
3                     0.83                      0.119   
4                    0.941                      0.077   
5                    0.957                      0.128   
6                      1.1                      0.068   
7                    1.116                      0.091   
8                    0.931                      0.072   
9                    0.947                      0.091   

  PREÇO MÍNIMO DISTRIBUIÇÃO PREÇO MÁXIMO DISTRIBUIÇÃO  \
0                    0.4201                    0.9666   
1                    0.5013                     01.05   
2                    0.5614                     1.161   
3                    0.5991                   1.22242   
4    

##Pré Análise dos Dados

In [336]:
#Verificando o tamanho do nosso dataset
df_base.shape

(120823, 18)

In [337]:
#Verificando as colunas existentes no dataset
df_base.columns

Index(['DATA INICIAL', 'DATA FINAL', 'REGIÃO', 'ESTADO', 'PRODUTO',
       'NÚMERO DE POSTOS PESQUISADOS', 'UNIDADE DE MEDIDA',
       'PREÇO MÉDIO REVENDA', 'DESVIO PADRÃO REVENDA', 'PREÇO MÍNIMO REVENDA',
       'PREÇO MÁXIMO REVENDA', 'MARGEM MÉDIA REVENDA',
       'COEF DE VARIAÇÃO REVENDA', 'PREÇO MÉDIO DISTRIBUIÇÃO',
       'DESVIO PADRÃO DISTRIBUIÇÃO', 'PREÇO MÍNIMO DISTRIBUIÇÃO',
       'PREÇO MÁXIMO DISTRIBUIÇÃO', 'COEF DE VARIAÇÃO DISTRIBUIÇÃO'],
      dtype='object')

In [338]:
COLUNAS = ['DATA INICIAL', 'DATA FINAL', 'REGIÃO', 'ESTADO', 'PRODUTO', 'NÚMERO DE POSTOS PESQUISADOS',
           'UNIDADE DE MEDIDA', 'PREÇO MÉDIO REVENDA', 'DESVIO PADRÃO REVENDA', 'PREÇO MÍNIMO REVENDA',
           'PREÇO MÁXIMO REVENDA', 'MARGEM MÉDIA REVENDA', 'COEF DE VARIAÇÃO REVENDA', 'PREÇO MÉDIO DISTRIBUIÇÃO',
           'DESVIO PADRÃO DISTRIBUIÇÃO', 'PREÇO MÍNIMO DISTRIBUIÇÃO', 'PREÇO MÁXIMO DISTRIBUIÇÃO',
           'COEF DE VARIAÇÃO DISTRIBUIÇÃO']

Não foi possível visualizar em df.head() se o dataset possui células vazias ou nulas. Para isto, iremos utilizar o comando isna(), isnull() e somar por coluna.

In [339]:
#Verificando se o dataset possui células:
#   * NaN se o tipo de dado for numérico
#   * None ou NaN se o tipo de dado for objeto
#   * NaT se o tipo de dado for data
df_base.isna().sum()

DATA INICIAL                     0
DATA FINAL                       0
REGIÃO                           0
ESTADO                           0
PRODUTO                          0
NÚMERO DE POSTOS PESQUISADOS     0
UNIDADE DE MEDIDA                0
PREÇO MÉDIO REVENDA              0
DESVIO PADRÃO REVENDA            0
PREÇO MÍNIMO REVENDA             0
PREÇO MÁXIMO REVENDA             0
MARGEM MÉDIA REVENDA             0
COEF DE VARIAÇÃO REVENDA         0
PREÇO MÉDIO DISTRIBUIÇÃO         0
DESVIO PADRÃO DISTRIBUIÇÃO       0
PREÇO MÍNIMO DISTRIBUIÇÃO        0
PREÇO MÁXIMO DISTRIBUIÇÃO        0
COEF DE VARIAÇÃO DISTRIBUIÇÃO    0
dtype: int64

In [340]:
#Verificando se o dataset possui células:
#   * None ou numpy.NaN.

#OBS: Células do tipo '' ou numpy.inf não são mapadas por essa função a menos que o comando abaixo seja rodado:
#pd.options.mode.use_inf_as_na = True
df_base.isnull().sum()

DATA INICIAL                     0
DATA FINAL                       0
REGIÃO                           0
ESTADO                           0
PRODUTO                          0
NÚMERO DE POSTOS PESQUISADOS     0
UNIDADE DE MEDIDA                0
PREÇO MÉDIO REVENDA              0
DESVIO PADRÃO REVENDA            0
PREÇO MÍNIMO REVENDA             0
PREÇO MÁXIMO REVENDA             0
MARGEM MÉDIA REVENDA             0
COEF DE VARIAÇÃO REVENDA         0
PREÇO MÉDIO DISTRIBUIÇÃO         0
DESVIO PADRÃO DISTRIBUIÇÃO       0
PREÇO MÍNIMO DISTRIBUIÇÃO        0
PREÇO MÁXIMO DISTRIBUIÇÃO        0
COEF DE VARIAÇÃO DISTRIBUIÇÃO    0
dtype: int64

In [341]:
#Verificando se o dataset possui células:
#   * None ou numpy.NaN ou '' ou nump.inf.
pd.options.mode.use_inf_as_na = True
df_base.isnull().sum()

DATA INICIAL                     0
DATA FINAL                       0
REGIÃO                           0
ESTADO                           0
PRODUTO                          0
NÚMERO DE POSTOS PESQUISADOS     0
UNIDADE DE MEDIDA                0
PREÇO MÉDIO REVENDA              0
DESVIO PADRÃO REVENDA            0
PREÇO MÍNIMO REVENDA             0
PREÇO MÁXIMO REVENDA             0
MARGEM MÉDIA REVENDA             0
COEF DE VARIAÇÃO REVENDA         0
PREÇO MÉDIO DISTRIBUIÇÃO         0
DESVIO PADRÃO DISTRIBUIÇÃO       0
PREÇO MÍNIMO DISTRIBUIÇÃO        0
PREÇO MÁXIMO DISTRIBUIÇÃO        0
COEF DE VARIAÇÃO DISTRIBUIÇÃO    0
dtype: int64

Todas as colunas não possuem valores nulos.

In [342]:
#Analisando a quantidade de valores únicos para as colunas
#'DATA INICIAL', 'DATA FINAL', 'REGIÃO', 'ESTADO', 'PRODUTO', 'UNIDADE DE MEDIDA'

for i in [0, 1, 2, 3, 4, 6]:
  coluna = df_base[COLUNAS[i]].unique()
  tamanho = len(coluna)
  print(f'{COLUNAS[i]} tem {tamanho} valores únicos')

DATA INICIAL tem 873 valores únicos
DATA FINAL tem 873 valores únicos
REGIÃO tem 5 valores únicos
ESTADO tem 27 valores únicos
PRODUTO tem 9 valores únicos
UNIDADE DE MEDIDA tem 3 valores únicos


In [343]:
df_base['REGIÃO'].value_counts()

NORDESTE        41671
NORTE           27989
SUDESTE         19270
CENTRO OESTE    17449
SUL             14444
Name: REGIÃO, dtype: int64

In [344]:
df_base['ESTADO'].value_counts()

PARAIBA                4819
RIO DE JANEIRO         4819
SAO PAULO              4819
RIO GRANDE DO SUL      4819
BAHIA                  4819
SANTA CATARINA         4818
RIO GRANDE DO NORTE    4818
MINAS GERAIS           4817
ESPIRITO SANTO         4815
PERNAMBUCO             4813
CEARA                  4813
PARANA                 4807
ALAGOAS                4806
SERGIPE                4795
MATO GROSSO DO SUL     4793
MATO GROSSO            4566
AMAZONAS               4414
GOIAS                  4134
PIAUI                  4017
MARANHAO               3971
DISTRITO FEDERAL       3956
PARA                   3955
TOCANTINS              3953
RONDONIA               3950
ACRE                   3948
RORAIMA                3938
AMAPA                  3831
Name: ESTADO, dtype: int64

Os dados estão bem distibuídos pelos estados.

In [345]:
df_base['PRODUTO'].value_counts()

GASOLINA COMUM        23570
GLP                   23561
ETANOL HIDRATADO      23440
ÓLEO DIESEL           21194
GNV                   14469
ÓLEO DIESEL S10        9113
OLEO DIESEL S10        2376
OLEO DIESEL            2351
GASOLINA ADITIVADA      749
Name: PRODUTO, dtype: int64

Precisa dar uma tratada na coluna PRODUTO.

In [346]:
df_base['UNIDADE DE MEDIDA'].value_counts()

R$/l       82793
R$/13Kg    23561
R$/m3      14469
Name: UNIDADE DE MEDIDA, dtype: int64

Tem que tomar cuidado quando for comparar

##Tratamento dos Dados

In [347]:
#Analisando os tipos de variáveis das colunas
df = df_base.copy()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120823 entries, 0 to 120822
Data columns (total 18 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   DATA INICIAL                   120823 non-null  object 
 1   DATA FINAL                     120823 non-null  object 
 2   REGIÃO                         120823 non-null  object 
 3   ESTADO                         120823 non-null  object 
 4   PRODUTO                        120823 non-null  object 
 5   NÚMERO DE POSTOS PESQUISADOS   120823 non-null  int64  
 6   UNIDADE DE MEDIDA              120823 non-null  object 
 7   PREÇO MÉDIO REVENDA            120823 non-null  float64
 8   DESVIO PADRÃO REVENDA          120823 non-null  float64
 9   PREÇO MÍNIMO REVENDA           120823 non-null  float64
 10  PREÇO MÁXIMO REVENDA           120823 non-null  float64
 11  MARGEM MÉDIA REVENDA           120823 non-null  object 
 12  COEF DE VARIAÇÃO REVENDA      

Note que algumas colunas que deveriam ser numéricas estão como object.

EXEMPLO:
* MARGEM MÉDIA REVENDA
* PREÇO MÉDIO DISTRIBUIÇÃO
* DESVIO PADRÃO DISTRIBUIÇÃO
* PREÇO MÍNIMO DISTRIBUIÇÃO
* PREÇO MÁXIMO DISTRIBUIÇÃO
* COEF DE VARIAÇÃO DISTRIBUIÇÃO

Será feita a conversão das mesmas.

In [348]:
#Convertendo para numérico (float neste caso) e coagindo quando necessário
LISTA_NOME_COLUNAS_CONVERTIDAS = ["MARGEM MÉDIA REVENDA", "PREÇO MÉDIO DISTRIBUIÇÃO", "DESVIO PADRÃO DISTRIBUIÇÃO",
                                  "PREÇO MÍNIMO DISTRIBUIÇÃO", "PREÇO MÁXIMO DISTRIBUIÇÃO", "COEF DE VARIAÇÃO DISTRIBUIÇÃO"]

for i in range(0, len(LISTA_NOME_COLUNAS_CONVERTIDAS)):
  nome = LISTA_NOME_COLUNAS_CONVERTIDAS[i]
  #O coerce foi usado para converter em 'NaN' os casos em que não foi possível fazer o cast.
  #O número a ser convertido não pode ser menor que (np.iinfo(np.int64).min)
  # e nem maior que (np.iinfo(np.uint64).max)
  df[nome] = pd.to_numeric(df[nome],errors='coerce')

In [349]:
pd.options.mode.use_inf_as_na = False
df.isna().sum()

DATA INICIAL                        0
DATA FINAL                          0
REGIÃO                              0
ESTADO                              0
PRODUTO                             0
NÚMERO DE POSTOS PESQUISADOS        0
UNIDADE DE MEDIDA                   0
PREÇO MÉDIO REVENDA                 0
DESVIO PADRÃO REVENDA               0
PREÇO MÍNIMO REVENDA                0
PREÇO MÁXIMO REVENDA                0
MARGEM MÉDIA REVENDA             3431
COEF DE VARIAÇÃO REVENDA            0
PREÇO MÉDIO DISTRIBUIÇÃO         3400
DESVIO PADRÃO DISTRIBUIÇÃO       3400
PREÇO MÍNIMO DISTRIBUIÇÃO        3400
PREÇO MÁXIMO DISTRIBUIÇÃO        3400
COEF DE VARIAÇÃO DISTRIBUIÇÃO    3400
dtype: int64

In [350]:
df.isnull().sum()

DATA INICIAL                        0
DATA FINAL                          0
REGIÃO                              0
ESTADO                              0
PRODUTO                             0
NÚMERO DE POSTOS PESQUISADOS        0
UNIDADE DE MEDIDA                   0
PREÇO MÉDIO REVENDA                 0
DESVIO PADRÃO REVENDA               0
PREÇO MÍNIMO REVENDA                0
PREÇO MÁXIMO REVENDA                0
MARGEM MÉDIA REVENDA             3431
COEF DE VARIAÇÃO REVENDA            0
PREÇO MÉDIO DISTRIBUIÇÃO         3400
DESVIO PADRÃO DISTRIBUIÇÃO       3400
PREÇO MÍNIMO DISTRIBUIÇÃO        3400
PREÇO MÁXIMO DISTRIBUIÇÃO        3400
COEF DE VARIAÇÃO DISTRIBUIÇÃO    3400
dtype: int64

In [351]:
pd.options.mode.use_inf_as_na = True
df.isnull().sum()

DATA INICIAL                        0
DATA FINAL                          0
REGIÃO                              0
ESTADO                              0
PRODUTO                             0
NÚMERO DE POSTOS PESQUISADOS        0
UNIDADE DE MEDIDA                   0
PREÇO MÉDIO REVENDA                 0
DESVIO PADRÃO REVENDA               0
PREÇO MÍNIMO REVENDA                0
PREÇO MÁXIMO REVENDA                0
MARGEM MÉDIA REVENDA             3431
COEF DE VARIAÇÃO REVENDA            0
PREÇO MÉDIO DISTRIBUIÇÃO         3400
DESVIO PADRÃO DISTRIBUIÇÃO       3400
PREÇO MÍNIMO DISTRIBUIÇÃO        3400
PREÇO MÁXIMO DISTRIBUIÇÃO        3400
COEF DE VARIAÇÃO DISTRIBUIÇÃO    3400
dtype: int64

In [352]:
df_base['MARGEM MÉDIA REVENDA'].value_counts()

-99999.0    5517
-           3431
0.361        329
0.326        317
0.333        313
            ... 
11.091         1
9.923          1
13.273         1
13.952         1
22.671         1
Name: MARGEM MÉDIA REVENDA, Length: 12885, dtype: int64

São exatamente 3431 linhas que nã puderam ser conertidas para numeric. Qual é o significado de '-'?

In [353]:
df_base['PREÇO MÉDIO DISTRIBUIÇÃO'].value_counts()

-99999.0    5508
-           3400
1.283         96
1.684         92
1.681         92
            ... 
34.843         1
39.918         1
32.663         1
28.48          1
51.166         1
Name: PREÇO MÉDIO DISTRIBUIÇÃO, Length: 17170, dtype: int64

São exatamente 3400 linhas que nã puderam ser conertidas para numeric. Qual é o significado de '-'?

In [354]:
df_base['DESVIO PADRÃO DISTRIBUIÇÃO'].value_counts()

-99999.0    5508
-           3400
0           3235
0.046       1108
0.056       1062
            ... 
3.676          1
7.233          1
5.469          1
5.516          1
5.442          1
Name: DESVIO PADRÃO DISTRIBUIÇÃO, Length: 6144, dtype: int64

São exatamente 3400 linhas que nã puderam ser conertidas para numeric. Qual é o significado de '-'?

In [355]:
df_base['PREÇO MÍNIMO DISTRIBUIÇÃO'].value_counts()

-99999.0    5508
-           3400
24           832
26           469
22.5         446
            ... 
1.8232         1
1.8542         1
2.1454         1
1.9356         1
2.9328         1
Name: PREÇO MÍNIMO DISTRIBUIÇÃO, Length: 23036, dtype: int64

São exatamente 3400 linhas que nã puderam ser conertidas para numeric. Qual é o significado de '-'?

In [356]:
df_base['PREÇO MÁXIMO DISTRIBUIÇÃO'].value_counts()

-99999.0    5508
-           3400
32           844
35           728
30           707
            ... 
40.699         1
1.2786         1
2.7602         1
42.4           1
3.8237         1
Name: PREÇO MÁXIMO DISTRIBUIÇÃO, Length: 24150, dtype: int64

São exatamente 3400 linhas que nã puderam ser conertidas para numeric. Qual é o significado de '-'?

In [357]:
df_base['COEF DE VARIAÇÃO DISTRIBUIÇÃO'].value_counts()

-99999.0    5508
-           3400
0.022       3304
0           3292
0.021       3284
            ... 
0.609          1
0.496          1
0.35           1
0.365          1
0.591          1
Name: COEF DE VARIAÇÃO DISTRIBUIÇÃO, Length: 400, dtype: int64

São exatamente 3400 linhas que nã puderam ser conertidas para numeric. Qual é o significado de '-'?

Perceba que agora existem varias células vazias. Precisamos tratá-las antes de continuarmos.

Analisando a data, seria interessante trabalhar com o dia, mês e ano em colunas separadas para facilitar a análise.

In [358]:
#Novas colunas com o dia, mês e ano baseadas na coluna "DATA INICIAL" e "DATA FINAL"

   #dia_inicial = pd.DatetimeIndex(df_base['DATA INICIAL']).day
   #dia_final = pd.DatetimeIndex(df_base['DATA FINAL']).day

   #mes_inicial = pd.DatetimeIndex(df_base['DATA INICIAL']).month
   #mes_final = pd.DatetimeIndex(df_base['DATA FINAL']).month

   #ano_inicial = pd.DatetimeIndex(df_base['DATA INICIAL']).year
   #ano_final = pd.DatetimeIndex(df_base['DATA FINAL']).year

LISTA_NOME_COLUNAS_NOVAS = ['DIA INICIAL', 'MES INICIAL', 'ANO INICIAL', 'DIA FINAL', 'MES FINAL', 'ANO FINAL']

lista_colunas = list()

for i in range(0, len(LISTA_NOME_COLUNAS_NOVAS)):
  nome = LISTA_NOME_COLUNAS_NOVAS[i]

  if 'INICIAL' in nome:
    coluna_aux = df['DATA INICIAL']
  elif 'FINAL' in nome:
    coluna_aux = df['DATA FINAL']
  else:
    print('Erro Inesperado')
    continue

  if 'DIA' in nome:
    coluna = pd.DatetimeIndex(coluna_aux).day
  elif 'MES' in nome:
    coluna = pd.DatetimeIndex(coluna_aux).month
  elif 'ANO' in nome:
    coluna = pd.DatetimeIndex(coluna_aux).year
  else:
    print('Erro Inesperado')
    continue

#Adicionando as colunas no DataFrame
#df_base = df_base.assign(dia_inicial = dia_inicial, mes_inicial = mes_inicial, ano_inicial = ano_inicial, dia_final = dia_final, mes_final = mes_final, ano_final = ano_final)

  df.insert(len(df.columns), LISTA_NOME_COLUNAS_NOVAS[i], coluna)

#Visualizando o dataset modificado
df.head()

DATA INICIAL  DATA FINAL        REGIÃO              ESTADO  \
0   2004-05-09  2004-05-15  CENTRO OESTE    DISTRITO FEDERAL   
1   2004-05-09  2004-05-15  CENTRO OESTE               GOIAS   
2   2004-05-09  2004-05-15  CENTRO OESTE         MATO GROSSO   
3   2004-05-09  2004-05-15  CENTRO OESTE  MATO GROSSO DO SUL   
4   2004-05-09  2004-05-15      NORDESTE             ALAGOAS   

            PRODUTO  NÚMERO DE POSTOS PESQUISADOS UNIDADE DE MEDIDA  \
0  ETANOL HIDRATADO                           127              R$/l   
1  ETANOL HIDRATADO                           387              R$/l   
2  ETANOL HIDRATADO                           192              R$/l   
3  ETANOL HIDRATADO                           162              R$/l   
4  ETANOL HIDRATADO                           103              R$/l   

   PREÇO MÉDIO REVENDA  DESVIO PADRÃO REVENDA  PREÇO MÍNIMO REVENDA  ...  \
0                1.288                  0.016                  1.19  ...   
1                1.162                  0.114                  0.89  ...   
2                1.389                  0.097                  1.18  ...   
3                1.262                  0.070                  1.09  ...   
4                1.181                  0.078                  1.05  ...   

   DESVIO PADRÃO DISTRIBUIÇÃO  PREÇO MÍNIMO DISTRIBUIÇÃO  \
0                       0.110                     0.4201   
1                       0.088                     0.5013   
2                       0.095                     0.5614   
3                       0.119                     0.5991   
4                       0.077                     0.7441   

   PREÇO MÁXIMO DISTRIBUIÇÃO  COEF DE VARIAÇÃO DISTRIBUIÇÃO  DIA INICIAL  \
0                    0.96660                          0.133            9   
1                    1.05000                          0.115            9   
2                    1.16100                          0.098            9   
3                    1.22242                          0.143            9   
4                    1.03170                          0.082            9   

   MES INICIAL  ANO INICIAL  DIA FINAL  MES FINAL  ANO FINAL  
0            5         2004         15          5       2004  
1            5         2004         15          5       2004  
2            5         2004         15          5       2004  
3            5         2004         15          5       2004  
4            5         2004         15          5       2004  

[5 rows x 24 columns]

In [359]:
#Analisando os tipos de variáveis das colunas
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120823 entries, 0 to 120822
Data columns (total 24 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   DATA INICIAL                   120823 non-null  object 
 1   DATA FINAL                     120823 non-null  object 
 2   REGIÃO                         120823 non-null  object 
 3   ESTADO                         120823 non-null  object 
 4   PRODUTO                        120823 non-null  object 
 5   NÚMERO DE POSTOS PESQUISADOS   120823 non-null  int64  
 6   UNIDADE DE MEDIDA              120823 non-null  object 
 7   PREÇO MÉDIO REVENDA            120823 non-null  float64
 8   DESVIO PADRÃO REVENDA          120823 non-null  float64
 9   PREÇO MÍNIMO REVENDA           120823 non-null  float64
 10  PREÇO MÁXIMO REVENDA           120823 non-null  float64
 11  MARGEM MÉDIA REVENDA           117392 non-null  float64
 12  COEF DE VARIAÇÃO REVENDA      

In [362]:
#Visualizando algumas estatísticas rápidas do dataset
df.describe()

NÚMERO DE POSTOS PESQUISADOS  PREÇO MÉDIO REVENDA  \
count                 120823.000000        120823.000000   
mean                     221.066461            11.473440   
std                      387.940239            19.161823   
min                        1.000000             0.766000   
25%                       39.000000             2.157000   
50%                       99.000000             2.899000   
75%                      226.000000             4.026000   
max                     4167.000000           107.500000   

       DESVIO PADRÃO REVENDA  PREÇO MÍNIMO REVENDA  PREÇO MÁXIMO REVENDA  \
count          120823.000000         120823.000000         120823.000000   
mean                0.684132              9.977821             13.233902   
std                 1.403317             16.433783             22.451445   
min                 0.000000              0.590000              0.999000   
25%                 0.075000              1.989000              2.420000   
50%                 0.121000              2.690000              3.200000   
75%                 0.204000              3.790000              4.589000   
max                10.748000            100.000000            120.000000   

       MARGEM MÉDIA REVENDA  COEF DE VARIAÇÃO REVENDA  \
count         117392.000000             120823.000000   
mean           -4697.376395                  0.045298   
std            21163.482182                  0.024763   
min           -99999.000000                  0.000000   
25%                0.272000                  0.029000   
50%                0.381000                  0.041000   
75%                0.554000                  0.058000   
max               36.847000                  0.395000   

       PREÇO MÉDIO DISTRIBUIÇÃO  DESVIO PADRÃO DISTRIBUIÇÃO  \
count             117423.000000               117423.000000   
mean               -4682.188208                -4690.133173   
std                21145.807211                21144.040032   
min               -99999.000000               -99999.000000   
25%                    1.757000                    0.044000   
50%                    2.359000                    0.076000   
75%                    3.364000                    0.148000   
max                   83.137000                   18.385000   

       PREÇO MÍNIMO DISTRIBUIÇÃO  PREÇO MÁXIMO DISTRIBUIÇÃO  \
count              117423.000000              117423.000000   
mean                -4683.304895               -4680.964830   
std                 21145.558284               21146.080296   
min                -99999.000000              -99999.000000   
25%                     1.638000                   1.900250   
50%                     2.238000                   2.537600   
75%                     3.200000                   3.575750   
max                    83.000000                  94.500000   

       COEF DE VARIAÇÃO DISTRIBUIÇÃO    DIA INICIAL    MES INICIAL  \
count                  117423.000000  120823.000000  120823.000000   
mean                    -4690.642465      15.713829       6.468876   
std                     21143.927009       8.806543       3.475266   
min                    -99999.000000       1.000000       1.000000   
25%                         0.019000       8.000000       3.000000   
50%                         0.029000      16.000000       6.000000   
75%                         0.058000      23.000000      10.000000   
max                         0.780000      31.000000      12.000000   

        ANO INICIAL      DIA FINAL      MES FINAL      ANO FINAL  
count  120823.00000  120823.000000  120823.000000  120823.000000  
mean     2012.74355      15.717123       6.469927    2012.759880  
std         4.88562       8.799908       3.472910       4.887389  
min      2004.00000       1.000000       1.000000    2004.000000  
25%      2009.00000       8.000000       3.000000    2009.000000  
50%      2013.00000      16.000000       6.000000    2013.000000  
75%      2017.00000      23.000